### PR TITLE
Resolve #2772: generate database flags (may omit global db)

### LINKED ISSUE #2772: Remove global DB restriction when converting PackageDbStack into ghc/ghc-pkg flags
In #2593, we removed the global DB restriction from `getInstalledPackages` for compilers that don't require this restriction. We should also change the function that converts the db stack into ghc/ghc-pkg flags (`D.S.Program.GHC.packageDBArgs`) accordingly. See this comment by @dcoutts: https://github.com/haskell/cabal/pull/2617#issuecomment-107149392.

### PR DESCRIPTION
**Pull request by phadej**
Alternative solution to https://github.com/haskell/cabal/pull/2776

### CODE REVIEW

**Comment by phadej** [comment]
ping @dcoutts, @23Skidoo

File: Cabal/Distribution/Simple/GHC.hs [hunk 1]
```diff
@@ -332,11 +332,10 @@ checkPackageDbEnvVar =
...
 checkPackageDbStack :: Compiler -> PackageDBStack -> IO ()
-checkPackageDbStack comp =
-  let ghcVersion = compilerVersion comp
-  in if ghcVersion < Version [7,6] []
-        then checkPackageDbStackPre76
-        else checkPackageDbStackPost76
+checkPackageDbStack comp = if flagPackageConf implInfo
+                              then checkPackageDbStackPre76
+                              else checkPackageDbStackPost76
+  where implInfo = ghcVersionImplInfo (compilerVersion comp)
```
Comment (23Skidoo) [inline Cabal/Distribution/Simple/GHC.hs]: We probably should do something to avoid recomputing `GhcImplInfo` each time it's needed.

Comment (23Skidoo) [inline Cabal/Distribution/Simple/GHC.hs]: Not in this patch, of course.

**Comment by 23Skidoo** [comment]
OK, I think I'll just merge it and a comment myself.

**Comment by 23Skidoo** [comment]
Merged, thanks!

**Comment by phadej** [comment]
Np, thanks for commening!

**Pull request merged**